addons/sale/models/sale_order_line.py

```diff
@@ -404,11 +414,16 @@ def _compute_product_uom_qty(self):
             if float_compare(product_uom_qty, line.product_uom_qty, precision_rounding=line.product_uom.rounding) != 0:
                 line.product_uom_qty = product_uom_qty
 
-    @api.depends('product_id')
+    @api.depends('product_id', 'is_downpayment', 'display_type')
     def _compute_product_uom(self):
         for line in self:
-            if not line.product_uom or (line.product_id.uom_id.id != line.product_uom.id):
-                line.product_uom = line.product_id.uom_id
+            if not line.product_uom or (line.product_id and line.product_id.uom_id.id != line.product_uom.id):
+                if line.display_type:
+                    line.product_uom = False
+                elif line.is_downpayment:
+                    line.product_uom = self.env.ref('uom.product_uom_unit')
```

Comment:
Do we even need an uom if there is no product ?
I guess it could make sense to hide uom & qties for dp lines.